### ISSUE TITLE
encoding/xml: Unmarshal does not properly handle namespace in XMLName field

### ISSUE THREAD

**Issue by fantasist** [golang/go#9775]
Example:
type A struct {
  XMLName xml.Name \`xml:"a:B"\`
}
Struct A is marshaled to "\<a:B\>\</a:B\>", as expected.
However if we run 
a := A{}
xml.Unmarshal([]byte("\<a:B\>\</a:B\>"), &a)

It results in an error saying "unexpected error: expected element type \<a:B\> but have \<B\>"

**Comment by c4milo** [comment]
This issue seems to be a duplicate of https://github.com/golang/go/issues/6800

**Comment by mikioh** [comment]
Dup of #6800.

**Comment by wookoouk** [comment]
@fantasist's example still fails in master

**Comment by rsc** [comment]
See #11841.

**Comment by rsc** [comment]
Blocked on #13400.

**Comment by iWdGo** [comment]
The provided example is invalid XML as the prefix `a` for element `<a:B>` is not bound. The prefix needs a declaration like xmlns:a="...". The reported error : `expected element type <a:B> but have <B>` points to the correct element.

**Comment by Demitroi** [comment]
@fantasist

Here's an unmarshal wrapper that allows namespace prefix

https://play.golang.org/p/bNe8Nl64U2b

**Comment by iwdgo** [comment]
The provided case written as a [test](https://go.dev/play/p/iSCSgRYmaUl?v=gotip) passes with the [suggested CL](https://go-review.googlesource.com/c/go/+/116056) of #9519 .
